--- a/src/solver/PCfrSolver.cpp
+++ b/src/solver/PCfrSolver.cpp
@@ -910,6 +910,7 @@
         shared_ptr<Trainable> trainable = one_node->getTrainable(deal,false);
         if(trainable != nullptr) {
             (*retval)["strategy"] = trainable->dump_strategy(false);
+            try { (*retval)["evs"] = trainable->dump_evs(); } catch(...) { }
             for(vector<int> one_exchange:exchange_color_list){
                 int rank1 = one_exchange[0];
                 int rank2 = one_exchange[1];
